### PR TITLE
fix #157 parsing error with null/true/false, jison 0.4.17

### LINKED ISSUE #157: white space significant in nools rule constraints, when it shouldn't be.  source of easy errors.
This constraint compiles/executes:

  dw : DimensionalWeight dw.value != null  && (dw.valid == null || dw.valid == false);

While this one does not:

dw : DimensionalWeight dw.value !=null  && (dw.valid == null || dw.valid == false);

The white space between the "!=" and the "null" has been removed.

### PR DESCRIPTION
**Pull request by DevSide**
fix #157 parsing error with null/true/false
In some case like `==null ` or `!=true` without whitespace, keywords were considered as an identifier instead of a value. 

- grammar updated for null/boolean cases
- add tests to cover null and no whitespace cases
- jison upgrade to 0.4.17

### CODE REVIEW

File: test/parser.test.js [hunk 1]
```diff
@@ -841,10 +841,7 @@ it.describe("A Parser", function (it) {
             "and",
             "AND",
             "or",
-            "OR",
-            "null",
```
Comment (DevSide) [inline test/parser.test.js]: null/boolean are no more considered as identifiers
